components/uuf-core/src/main/java/org/wso2/carbon/uuf/api/config/ComponentManifest.java

```diff
@@ -0,0 +1,207 @@
+     */
+    public void setBindings(List<Binding> bindings) {
+        this.bindings = bindings;
+    }
+
+    /**
+     * Represent an API entry in the component manifest file in an UUF Component.
+     *
+     * @since 1.0.0
+     */
+    public static class API {
+
+        private String className;
+        private String uri;
+
+        /**
+         * Returns the name of the class of this API.
+         *
+         * @return name of the class of this API
+         */
+        public String getClassName() {
+            return className;
+        }
+
+        /**
+         * Sets the name of the class of this API.
+         *
+         * @param className name of the class to be set
+         */
+        public void setClassName(String className) {
+            this.className = className;
+        }
+
+        /**
+         * Returns the URI of this API.
+         *
+         * @return URI of this API
+         */
+        public String getUri() {
+            return uri;
+        }
+
+        /**
+         * Sets the URI of this API.
+         *
+         * @param uri URI to be set
+         */
+        public void setUri(String uri) {
+            this.uri = uri;
+        }
+    }
+
+    /**
+     * Represent a binding entry in thr component manifest file in an UUF Component.
+     *
+     * @since 1.0.0
+     */
+    public static class Binding {
+
+        /**
+         * Binding combine mode prepend.
+         */
+        public static final String MODE_PREPEND = "PREPEND";
+        /**
+         * Binging combine mode append.
+         */
+        public static final String MODE_APPEND = "APPEND";
+        /**
+         * Binding combine mode overwrite.
+         */
+        public static final String MODE_OVERWRITE = "OVERWRITE";
+
+        private String zoneName;
+        private String mode;
+        private List<String> fragments;
+
+        /**
+         * Returns the zone name of this binding.
+         *
+         * @return zone name of this binding
+         */
+        public String getZoneName() {
+            return zoneName;
+        }
+
+        /**
+         * Sets the zone name of this binding.
+         *
+         * @param zoneName zone name to be set
+         */
+        public void setZoneName(String zoneName) {
+            this.zoneName = zoneName;
+        }
+
+        /**
+         * Returns the mode of this binding.
+         *
+         * @return mode of this binding
+         */
+        public String getMode() {
+            return mode;
+        }
+
+        /**
+         * Sets the mode of this binding.
+         *
+         * @param mode mode to be set
+         * @exception IllegalArgumentException if the {@code mode} is not {@link #MODE_PREPEND} or {@link #MODE_APPEND}
+         *                                     or {@link #MODE_OVERWRITE}
+         * @see #MODE_PREPEND
+         * @see #MODE_APPEND
+         * @see #MODE_OVERWRITE
+         */
+        public void setMode(String mode) {
```

Comment:
on the config side, let's give only one option. let's go with simple for the config side eg ; append